packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -152,45 +172,33 @@ export class MarketOperationUtils {
             _opts,
         );
 
-        const balancerPromise = DexOrderSampler.ops
-            .getSellQuotesAsync(
-                difference([ERC20BridgeSource.Balancer], _opts.excludedSources),
-                makerToken,
-                takerToken,
-                sampleAmounts,
-                this._wethAddress,
-                this._sampler.balancerPoolsCache,
-                this._liquidityProviderRegistry,
-                this._multiBridge,
-                this._sampler.bancorService,
-            )
-            .then(async r => this._sampler.executeAsync(r));
+        const offChainBalancerPromise = sampleBalancerOffChain
+            ? this._sampler.getBalancerSellQuotesAsync(makerToken, takerToken, sampleAmounts)
+            : Promise.resolve([]);
+
+        const offChainBancorPromise = _opts.excludedSources.includes(ERC20BridgeSource.Bancor)
+            ? Promise.resolve([])
+            : this._sampler.getBancorSellQuotesAsync(makerToken, takerToken, sampleAmounts);
```

Comment:
To be safe (I forgot to exclude this). If it's not excluded, we should just perform one sample (2 requests). Until we can sample efficiently.
```suggestion
            : this._sampler.getBancorSellQuotesAsync(makerToken, takerToken, [takerAmount]);
```